test_data/asm/block_to_block_empty_submachine.asm

```diff
@@ -0,0 +1,54 @@
+// This test is mostly a copy of block_to_block.asm, but:
+// - The machines have dynamic length (either 4 or 8)
+// - `instr_add` can be set by the prover.
+//
+// We set `instr_add` to zero in all rows at runtime. This simulates
+// a secondary machine that is never called (and, because it depends on
+// runtime information, it's not removed by the optimizer).
```

Comment:
Actually, the optimizer does not remove the machine if I set `instr_add` to zero, but it should!